rust/kcl-lib/src/execution/exec_ast.rs

```diff
@@ -1699,6 +1761,13 @@ impl Node<BinaryExpression> {
         Ok(value)
     }
 
+    fn missing_result_error(node: &Node<BinaryExpression>) -> KclError {
+        KclError::new_internal(KclErrorDetails::new(
+            "Internal error: missing result while evaluating binary expression".to_owned(),
```

Comment:
We shouldn't need to include the type in the message. The type is prefixed to the message everywhere, as far as I know.

```suggestion
            "Missing result while evaluating binary expression".to_owned(),
```